chain/bitcoin/address.go

```diff
@@ -37,11 +38,12 @@ func NewAddressEncoder(params *chaincfg.Params) AddressEncoder {
 
 // EncodeAddress implements the address.Encoder interface
 func (encoder AddressEncoder) EncodeAddress(rawAddr address.RawAddress) (address.Address, error) {
+	// Validate that the base58 address is in fact in correct format.
 	encodedAddr := base58.Encode([]byte(rawAddr))
-	if _, err := btcutil.DecodeAddress(encodedAddr, encoder.params); err != nil {
-		// Check that the address is valid.
+	if _, err := btcutil.DecodeAddress(encodedAddr, &chaincfg.RegressionNetParams); err != nil {
```

Comment:
Why are we always using the regression network?

Reply:
Oh! That's a bad mistake. Fixed it! Thanks!